server/src/db/catalog.rs

```diff
@@ -117,6 +133,72 @@ impl Catalog {
             .cloned()
             .context(UnknownPartition { partition_key })
     }
+
+    /// Return the chunks for specific partition
+    pub fn chunks(&self, partition_key: &str) -> Vec<Arc<DBChunk>> {
```

Comment:
```suggestion
    pub fn db_chunks(&self, partition_key: &str) -> Vec<Arc<DBChunk>> {
```

I think this may be confusing as a `DBChunk` is not the same as a `catalog::Chunk`

I personally think this function might be better if left on `Db` so as to avoid the use of `DBChunk` at all inside the catalog. However, that is just a stylistic preference I see no practical reason that this structure is a problem

Reply:
> I personally think this function might be better if left on Db so as to avoid the use of DBChunk

How do you feel about the change in general then, given the SchemaProvider gives out DBChunks?